src/Lumper.UI/Updater/Updater.cs

```diff
@@ -0,0 +1,420 @@
+
+        }
+        process = Process.Start(processInfo);
+    }
+    /// <summary>
+    /// Grab the JSON from the Github API and deserializes it
+    /// </summary>
+    /// <returns></returns>
+    /// <exception cref="Exception"></exception>
+    private static async Task<GithubUpdate> GetGithubUpdates()
+    {
+        var client = new HttpClient();
+
+        var request = new HttpRequestMessage() {
+            RequestUri = new Uri("https://api.github.com/repos/momentum-mod/lumper/releases"),
+            Method = HttpMethod.Get
+        };
+        client.DefaultRequestHeaders.Add("User-Agent", "Other");
+        HttpResponseMessage response = await client.SendAsync(request);
+
+        if (response.IsSuccessStatusCode)
+        {
+            var jsonString = await response.Content.ReadAsStringAsync();
+
+            return JsonConvert.DeserializeObject<GithubUpdate>(JsonNode.Parse(jsonString)[0].ToString());
+        }
+        else
+        {
+            throw new Exception("Could not connect - error " + response.StatusCode);
+        }
+    }
+    /// <summary>
+    /// Checks for possible update on the Github releases page by the tag name
+    /// </summary>
+    /// <returns>The current latest version, or null if it the latest</returns>
+    public static async Task<Version?> CheckForUpdate()
+    {
+        GithubUpdate assets;
+        try
+        {
+            assets = await GetGithubUpdates();
+        }
+        catch (Exception ex)
+        {
+            ButtonResult result = await MessageBoxManager
+            .GetMessageBoxStandard(
+                "Error",
+                ex.Message, ButtonEnum.Ok)
+            .ShowWindowDialogAsync(Program.Desktop.MainWindow);
+            return null;
+        }
+
+        //parse tag name to find the current and latest version
+        //finding the format of xx.yy.zz
+        Version? current;
+        Version? latest;
+        try
+        {
+            current = GetVersionFromString(Assembly.GetExecutingAssembly().GetName().Version.ToString());
+            latest = GetVersionFromString(assets.TagName);
+        }
+        catch (Exception ex)
+        {
+            ButtonResult result = await MessageBoxManager
+            .GetMessageBoxStandard(
+                "Error",
+                "Could not parse version number.", ButtonEnum.Ok)
+            .ShowWindowDialogAsync(Program.Desktop.MainWindow);
+            return null;
+
+        }
+        if (current != latest)
+            return latest;
+        else
+            return null;
+    }
+    public static async Task<Stream?> HttpDownload(string url, string fileName)
+    {
+        IoProgressWindow? progressWindow = null;
+        var buffer = System.Buffers.ArrayPool<byte>.Shared.Rent(80 * 1024);
+        var cts = new CancellationTokenSource();
+        var handler = new IoHandler(cts);
+
```

Comment:
We're doing other work after downloading, unzipping the file and moving it. So we should create the IoProgressWindow and related classes outside of this method, then pass them in. Then during this method push the progress bar along to 80% or so, then do the final 20% during the unzipping.